### ISSUE TITLE
Issue in Medusa V2 Learning Resources - Unit Price cannot be added during Cart Creation but document says so

### ISSUE THREAD

**Issue by sidd-kulk** [medusajs/medusa#8776]
### What Medusa version and documentation are you using?

v2

### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/medusajs/medusa/issues

### Issue Summary

In the V2 documentation mentioned below:
https://docs.medusajs.com/v2/resources/commerce-modules/cart
https://docs.medusajs.com/v2/api/store#carts_postcarts

It shows that Cart accepts Unit Price for an item, but as per discussion with team, it does not.  The documentation needs to be updated.  

### How can this issue be resolved?

1. Access the documentation as mentioned in the summary above.
2. Access the cart payload
3. It shows Unit Price under `items` node.

### Are you interested in working on this issue?

- [X] I would like to fix this issue

**Comment by shahednasser** [comment]
Thank you for opening this issue @sidd-kulk ! I've opened the PR #8820 to fix the request type of the API route, which will fix the API reference once the OpenAPI Specs are generated.

Regarding the example in https://docs.medusajs.com/v2/resources/commerce-modules/cart, I've checked with the team, and providing the `unit_price` is accepted for the module's main service method. so, no changes will be made there.
